source/fe/mapping_fe.cc

```diff
@@ -909,12 +909,78 @@ MappingFE<dim, spacedim>::transform_real_to_unit_cell(
   const typename Triangulation<dim, spacedim>::cell_iterator &cell,
   const Point<spacedim> &                                     p) const
 {
-  Assert(false, StandardExceptions::ExcNotImplemented());
+  const std::vector<Point<spacedim>> support_points =
+    this->compute_mapping_support_points(cell);
+
+  const double       eps        = 1.e-12 * cell->diameter();
+  const unsigned int loop_limit = 10;
+
+  Point<dim> p_unit;
+
+  unsigned int loop = 0;
+
+  // This loop solves the following problem:
+  // grad_F^T residual + (grad_F^T grad_F + grad_F^T hess_F^T dp) dp = 0
```

Comment:
```suggestion
  // grad_F^T residual + (grad_F^T grad_F + grad_F^T hess_F^T) dp = 0
```

Reply:
No. This was correct. I'm using  `grad_F + hess_F^T dp` as a higher order approximation of `grad_F`.